src/pybind/mgr/dashboard/services/rbd.py

```diff
@@ -245,10 +246,22 @@ def __call__(self, offset, length, exists):
         return total_used_size, snap_map
 
     @classmethod
-    def _rbd_image(cls, ioctx, pool_name, namespace, image_name):
+    def _rbd_image(cls, ioctx, pool_name, namespace, image_name):  # pylint: disable=R0912
         with rbd.Image(ioctx, image_name) as img:
-
             stat = img.stat()
+            snapshot_mode = img.mirror_image_get_mode() == rbd.RBD_MIRROR_IMAGE_MODE_SNAPSHOT
+            if img.mirror_image_get_mode() == rbd.RBD_MIRROR_IMAGE_MODE_JOURNAL:
+                stat['mirror_mode'] = 'journal'
+            elif snapshot_mode:
+                stat['mirror_mode'] = 'snapshot'
+                schedule_status = json.loads(mgr.remote(
+                    'rbd_support', 'mirror_snapshot_schedule_status')[1])
```

Comment:
`remote` calls require extra caution. What if the rbd_support module is not loaded (I know it's an always-on but other always-on modules can be disabled)? How are we handling that situation? This might break a service that previously didn't depend on another module.

Reply:
There seems to be 2 possible errors, I've treated those errors in case they happen.